### ISSUE TITLE
What to do if we are one Yalu b3 ?

### ISSUE THREAD

**Issue by papapalmlee** [kpwn/yalu102#71]
Can we install Yalu102 right away or do we have to restore first (iPhone 7Plus 10.1.1 + Yalu b3)

**Comment by Amcio** [comment]
i installed it right away on my SE and no problems.

**Comment by rpalmieri99** [comment]
7 is not supported yet. So stay on b3

**Comment by papapalmlee** [comment]
On the app it's clearly say it does " for 10.0/10.1 on iPhone7"

**Comment by papapalmlee** [comment]
I tried install over b3 and it says "failed, retry"

**Comment by Mila432** [comment]
@papapalmlee it clearly says "yalu102 for 6s/SE and iPad Pro 10.0/10.1/10.2"

**Comment by papapalmlee** [comment]
![img_9169](https://cloud.githubusercontent.com/assets/25366423/22332676/ad8ad36e-e404-11e6-8bd0-a874ac484d8a.PNG)

Well, Check this out

**Comment by HoldMyL** [comment]
Ignore that part. Luca says it's only compatible with 6s atm. Waiting for info on what to do with b3 as well

**Comment by appache7-2** [comment]
I will try for iphone 6, but the code could have been better commented .. maybe tracks on modules missing?

**Comment by beyondxx3** [comment]
I try. but...
/Users/Name/Documents/yalu102/yalu102/jailbreak.m:14:9: 'IOKit/IOKitLib.h' file not found

**Comment by 0x0luke** [comment]
The JB doesn't work yet on 6 and 6+, it causes a kernel panic or device freeze, remember that this project hasn't been completed yet and therefore not all devices have been added/supported. 

Also if you're missing IOKit you need to add it into the project in the build phases or find the files yourself. You need all the IOKit files if you're going to add them to the project like I did. 

Sources: I have the app on my iPhone 6+, 10.2.

**Comment by appache7-2** [comment]
Yep .. IOKitLib.h is missing to project ..

**Comment by 0x0luke** [comment]
IOKit isn't missing, it's a framework you need to add manually in the build phases

**Comment by 4taras4** [comment]
@beyondxx3  
Download the zip of the project (https://github.com/kennytm/iphone-private-frameworks/tree/master/IOKit), then place the header files in a folder that you might use for other projects.
Go to your target’s Build Settings. In the search bar write “headers”. In the results you’ll see a section called Search Paths. Change “Always Search User Paths” to YES, and add the path where you placed your headers in “User Header Search Paths”. Make sure to put the folder that contains the IOKit folder, and not the IOKit folder itself.
IOKit imports a file not present in these headers, head here to obtain it. **Place it in a folder called libkern, inside the Headers folder that you created.**

**Comment by tangalbert919** [comment]
I was too slow. I was just about to say that!
Great minds think alike. :smile:

**Comment by tangalbert919** [comment]
I can't add more offsets, but I can make the code more readable right now through the use of offsets, and I'll see if I can get 4k device support working. Given that I know nothing about Objective-C (except "uint_64" is 64-bit unsigned), me doing my best should help Luca get a head start in testing.
@kpwn, we all love you and your work, but please add the "Anyone But Stefan Esser" license so he can't use this.

**Comment by appache7-2** [comment]
I think Lucas has finished with the JB, he does not want to work on it anymore. Hence the public

**Comment by cawk** [comment]
Not at all he has stated already that he can't look at 4K devices until the weekend since he doesn't have one at the moment so stop talking shit

**Comment by mwoolweaver** [comment]
Link for proof

https://github.com/kpwn/yalu102/issues/28#issuecomment-275305581

**Comment by PaulBGD** [comment]
Does this support the 5S as well, since it's a 64bit phone?

**Comment by KILLCAMPER** [comment]
@PaulBGD Please read the read me in this link 
https://github.com/kpwn/yalu102

**Comment by PaulBGD** [comment]
@KILLCAMPER I'm not new to github, I checked the README and there's no information regarding list of devices that work or rules for contributing. Not everyone reads through 1k+ comment threads to figure out what hardware a project supports.

**Comment by KILLCAMPER** [comment]
There not going that old so they say less you want to contribute the offsets . And if that have a device to test the off sets in memory

**Comment by danielkoller** [comment]
I've used the 10.1.1 Yalu B3 and just updated my iPhone 6S Plus to 10.2. Used Cydia Impactor and the .ipa and everything is working quite stable so far. Had only one reboot yet.

**Comment by kpwn** [comment]
Linking to IOKit and stuff is not allowed. Closing.

Yalu102 is compatible with 10.1.1, even on previously jailbroken devices, but I suggest to remove the /.installed_yaluX file before updating.

**Comment by pwned4** [comment]
whenever i try to add the iokit.dlyib to my project it still doesn't recognize its there and i still get a iokit not found error. i don't understand your explanation on how to add the iokit from the GitHub folder, i add it through my current files already on my mac in the system/library/framworks folder. i try to copy the dylib from there into my project but no luck. BTW I'm running it onto my 5s 10.2
